### DISCUSSION TITLE
Strimzi managed pod's labels [app.kubernetes.io/version and app.kubernetes.io/component] not reflected

### DISCUSSION THREAD

**Discussion by palpandy** [strimzi/strimzi-kafka-operator#10349] [General]
Hello, 

I am trying to add app.kubernetes.io/version and app.kubernetes.io/component under labels in our kafka cluster and mirrormaker2 CRDs. I have added it inside the pod spec for each objects [Kafka,Zookeeper,Kafkaexporter,mirrormaker2] . But the changes are not reflected in the pod after the deployment. I added the other type of labels which is getting reflected but not the above said labels. And the common pattern I see these components are managed by strimzi. Can anyone tell me if that's a strimzi limitation or any other reason ? Please help me on this. 
I am using strimzi 0.40.0 and kafka 3.7.0

**Comment by scholzj** [comment]
At lest some of these labels are set by Strimzi and cannot be customized: https://strimzi.io/docs/operators/latest/full/deploying.html#ref-kubernetes-labels-str

**Reply by palpandy** [reply]
Hi scholzj, 

Thanks for replying. Yes, I am aware of the labels mentioned in the doc that you shared are set by strimzi. But, I am trying to add these labels [app.kubernetes.io/version and app.kubernetes.io/component] which is not mentioned in that doc. 
Is it not possible to add these labels [app.kubernetes.io/version and app.kubernetes.io/component] to the kafka / any pods ?  [managed by strimzi]

**Reply by scholzj** [reply]
Hmm, I thought `component` was there. Have you tried to set them on the custom resource?

**Reply by palpandy** [reply]
Yes, I set them on the custom resource and deployed through helm chart. So, in CRD it is present but it is not present in the pods [kafka,zk,mm2 etc] once it is deployed.

**Reply by scholzj** [reply]
Sorry, I needed to refresh my memory. These are excluded by default by `STRIMZI_LABELS_EXCLUSION_PATTERN`. You can change it by defining it as an environment variable in the Cluster Operator deployment:
* https://github.com/strimzi/strimzi-kafka-operator/blob/e8a590899a1f4d5986d0685ade4209f79c82c0e9/operator-common/src/main/java/io/strimzi/operator/common/model/Labels.java#L110
* https://strimzi.io/docs/operators/latest/full/deploying.html#ref-operator-cluster-str

**Reply by palpandy** [reply]
Ok sure. Let me try that. Thanks

**Reply by palpandy** [reply]
Hi Scholzj, 
I tried setting as an environment variable but no luck. In the [document](https://strimzi.io/docs/operators/latest/full/deploying.html#ref-operator-cluster-str:~:text=value1%2Clabel2%3Dvalue2-,STRIMZI_LABELS_EXCLUSION_PATTERN,-Optional%2C%20default%20regex) - , I see this line "The labels exclusion filter is not applied to labels in template sections such as spec.kafka.template.pod.metadata.labels." 
I am adding those labels in spec.kafka.template.pod.metadata.labels only so I guess this will not work though I set the exclusion pattern as an env variable because the filter is not applied to labels inside spec.kafka.template.pod.metadata. Please correct me If I am wrong.

**Reply by scholzj** [reply]
Oh, sorry. I guess I have not been exact enough. This applies to the labels from the custom resource. So changing it should allow you to set the labels on the `Kafka` CR etc. You can never specify these Kubernetes labels in the template section.

**Reply by palpandy** [reply]
Thanks for the clarification. So, there is no way to achieve my requirement I guess.

**Reply by scholzj** [reply]
As I said, you can use the environment variable \nd then set your labels on the custom resource. They will copied from there to the pods automatically.

**Reply by palpandy** [reply]
It worked as you said. Thank you so much for the support

**Comment by palpandy** [comment]
Resolved the issue
